### DISCUSSION TITLE
webpackJsonpCallback is not called when I navigate to a new page

### DISCUSSION THREAD

**Discussion by MarioSimou** [webpack/webpack#17543] [Q&A]
I am facing a weird issue when I try to load new chunks while navigating to a new page, which I am not sure what's the root cause of it. I have implemented code splitting in my app and the build runs okay, the chunks are created and the app loads fine on the initial load. However, when I navigate to a new page, the app fails with a `ChunkLoad` error. Initially, I thought that the location of the chunks was invalid, which I checked, but it seems alright. The HTTP calls on the network tab also show a 200 status code. I need to mention that the app runs on dev mode, which cancels any scenario of a caching issue.

I spent a bit of time reading the build code and I noticed that `webpackJsonpCallback` is never called when I move to a new page, which is not the case when the app loads for a first time. If I understood correctly, `webpackJsonpCallback` is responsible for registering a chunk in the modules and `installedChunks` stores. If the chunk doesn't exists in `installedChunks`, it is marked as failure by the `loadingEnded` callback, which runs as soon as the chunk loading is done. `__webpack_require__.f.j` is the method responsible for creating the JSONP script, injecting it to the DOM and attaching the `loadingEnded` callback.

Can anyone think of reason why `webpackJsonpCallback` is never called when I navigate to a new page?

**Comment by MarioSimou** [comment]
Okay, it seems that the issue is related to `output.globalObject`. By default, webpack uses `self` when it creates each chunk, which in my case it was referring to a different `self` on the initial load and when I was navigating to a new page. Setting `output.globalObject` to `globalThis` seems to fix my issue.

**Reply by amir-nazdaq** [reply]
I'm facing a similar issue, can you tell me please which files exactly you updated?

**Reply by MarioSimou** [reply]
Try to update [this](https://webpack.js.org/configuration/output/#outputglobalobject) option in your webpack configuration
